src/main/java/pl/tymoteuszboba/emojimize/parser/EmojiParser.java

```diff
@@ -6,24 +6,33 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class EmojiParser {
 
     public static String parse(VendorApi api, String originalString) {
-        String[] splittedPhrase = originalString.split(" ");
-
-        for (String word : splittedPhrase) {
-            if (word.startsWith(":") && word.endsWith(":")) {
-                String wordCopy = word;
-                word = word.replace(":", "");
-
-                try {
-                    Optional<Emoji> emoji = api.parseEmoji(word);
-                    if (emoji.isPresent()) {
-                        originalString = originalString.replace(wordCopy,
-                                new String(emoji.get().getUnicode().getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8));
-                    }
-                } catch (IOException ignored) {}
+        String[] splitPhrase = originalString.split(" ");
+
+        for (String word : splitPhrase) {
+            if (word.contains(":")) {
+                Matcher emojiMatcher = Pattern.compile("(?:[:].*?[:])").matcher(word);
+
+                while (emojiMatcher.find()) {
+                    int firstEmojiIndex = emojiMatcher.start();
+                    int lastEmojiIndex = emojiMatcher.end();
```

Comment:
I feel like these should be renamed as 'emoji**Left**Boundary' and 'emoji**Right**Boundary', respectively, not to confuse these indices with actual individual emojis